diff --git a/config/auto-aux/hasgot b/config/auto-aux/hasgot
index 54281a42da59..380aaaf094fc 100755
--- a/config/auto-aux/hasgot
+++ b/config/auto-aux/hasgot
@@ -15,7 +15,14 @@
 #*                                                                        *
 #**************************************************************************
 
-opts=""
+(echo "int main() { return 0; }") > hasgot.c
+$cc -Wno-implicit-function-declaration hasgot.c > /dev/null 2>/dev/null
+if test $? -eq 0 ; then
+  opts="-Wno-implicit-function-declaration";
+else
+  opts=""
+fi
+
 libs="$cclibs"
 args=$*
 rm -f hasgot.c
